packages/next/src/client/components/router-reducer/reducers/server-action-reducer.ts

```diff
@@ -290,7 +297,7 @@ export function serverActionReducer(
     },
     (e: any) => {
       // When the server action is rejected we don't update the state and instead call the reject handler of the promise.
-      reject(e)
+      reject(e instanceof Error ? e.message : e)
```

Comment:
Do we need to only reject with the message or it's fine to reject the error?

Reply:
I’m not sure, I think this is ok, but I might change this instanceof check to be more specific to the error that I throw in the fetch handler, just to avoid any change in behavior for other errors.